MapboxAndroidSDK/src/androidTest/java/com/spatialdev/osm/model/OSMColorConfigTest.java

```diff
@@ -0,0 +1,302 @@
+
+        OSMElement osmElement = new OSMNode(new LatLng(0.222222, 36.222222), colorConfig);
+        Context context = InstrumentationRegistry.getContext();
+        Drawable originalDrawable = context.getResources().getDrawable(R.mipmap.maki_star_orange_1);
+
+        //test the focus out drawable
+        osmElement.addOrEditTag("test_tag", "val_3");
+
+        Drawable focusOutDrawable = OSMColorConfig.getFocusOutDrawable(osmElement, context.getResources().getDrawable(R.mipmap.maki_star_orange_2));
+        OSMColorConfig.ARGB val3ArgbFocused = new OSMColorConfig.ARGB(val3Color);
+        OSMColorConfig.ARGB val3Argb = new OSMColorConfig.ARGB(
+                val3ArgbFocused.a - OSMColorConfig.FOCUS_OUT_ALPHA_DELTA,
+                val3ArgbFocused.r,
+                val3ArgbFocused.g,
+                val3ArgbFocused.b
+        );
+        Drawable val3Drawable = OSMColorConfig.applyColorFilterToDrawable(context.getResources().getDrawable(R.mipmap.maki_star_orange_3), val3Argb);
+
+        compareDrawableColorFilters(focusOutDrawable, originalDrawable, false);
+        compareDrawableColorFilters(focusOutDrawable, val3Drawable, true);
+    }
+
+    /**
+     * This method tests whether a non-default OSMColorConfig produces the right drawables
+     */
+    @Test
+    public void testNonDefaultConfigDrawable_unknownValue() {
+        HashMap<String, String> colorValues = new HashMap<>();
+        String defaultColor = "#532343";
+        String val1Color = "#324f21";
+        String val2Color = "#4a4f21";
+        String val3Color = "#ff4f25";
+        colorValues.put("val_1", val1Color);
+        colorValues.put("val_2", val2Color);
+        colorValues.put("val_3", val3Color);
+        colorValues.put("", defaultColor);//to be used for all other values apart from val_1, val_2, & val_3
+        OSMColorConfig colorConfig = new OSMColorConfig(true, "test_tag", colorValues);
+
+        OSMElement osmElement = new OSMNode(new LatLng(0.222222, 36.222222), colorConfig);
+        Context context = InstrumentationRegistry.getContext();
+        Drawable originalDrawable = context.getResources().getDrawable(R.mipmap.maki_star_orange_1);
+
+        //test the default drawable for unknown values
+        osmElement.addOrEditTag("test_tag", "fddfsdfds");
+
+        Drawable unknownDrawable = OSMColorConfig.getFocusInDrawable(osmElement, context.getResources().getDrawable(R.mipmap.maki_star_orange_2));
+        Drawable defaultDrawable = OSMColorConfig.applyColorFilterToDrawable(context.getResources().getDrawable(R.mipmap.maki_star_orange_3), new OSMColorConfig.ARGB(defaultColor));
+
+        compareDrawableColorFilters(unknownDrawable, originalDrawable, false);
+        compareDrawableColorFilters(unknownDrawable, defaultDrawable, true);
+    }
+
+    private void compareDrawableColorFilters(Drawable a, Drawable b, boolean checkEqual) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```

Comment:
It's a test that can only be done on some devices. Hence the reason for isolating it on its own function. I still haven't found a way of testing it on all devices